packages/vscode-dependencies-validation/src/autofix/packageJsonFileWatcher.ts

```diff
@@ -0,0 +1,30 @@
+import type { Uri } from "vscode";
+import { debounce } from "lodash";
+import { VscodeFileEventConfig } from "../vscodeTypes";
+import { handlePackageJsonEvent } from "./eventUtil";
+
+const debouncedHandlePackageJsonEvent = debounce(handlePackageJsonEvent, 3000);
+
+// TODO: what should happen after git clone ??
+export function addPackageJsonFileWatcher(
+  vscodeConfig: VscodeFileEventConfig
+): void {
+  const fileWatcher =
+    vscodeConfig.workspace.createFileSystemWatcher("**/package.json");
```

Comment:
the glob pattern seems to have some negate options.
perhaps that can be used to ignore package.json files inside node_modules
without needing a post-processing filter?
 - https://code.visualstudio.com/api/references/vscode-api#GlobPattern

Reply:
it is not supported. https://github.com/microsoft/vscode/issues/27051